src/KlarnaOnsiteMessaging.php

```diff
@@ -0,0 +1,147 @@
+<?php
+namespace Krokedil\KlarnaOnsiteMessaging;
+
+use Krokedil\KlarnaOnsiteMessaging\Pages\Product;
+use Krokedil\KlarnaOnsiteMessaging\Pages\Cart;
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+define( 'KOSM_VERSION', '0.0.1' );
+
+/**
+ * The orchestrator class.
+ */
+class KlarnaOnsiteMessaging {
+	/**
+	 * The internal settings state.
+	 *
+	 * @var Settings
+	 */
+	public $settings;
+
+	/**
+	 * Class constructor.
+	 *
+	 * @param array $settings Any existing KOSM settings.
+	 */
+	public function __construct( $settings ) {
+		$this->settings = new Settings( $settings );
+		$page           = new Product( $this->settings );
+		$cart           = new Cart( $this->settings );
+		$shortcode      = new Shortcode();
```

Comment:
Make all of these class properties that are private and accessible through a method in the class. That way no one can override them, but they are still accessible. This will also let people unhook actions if they dont want to run them at the same time we do for example